dali/operators/audio/nonsilence_op.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/static_switch.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(NonsilenceRegion)
+                .DocStr(R"code(The operator performs leading and trailing silence detection in an audio buffer.
+This operators' behaviour can be described as::
+
+  def nonsilence(buffer, cutoff_value, top_db, ref_db):
```

Comment:
I would rather not explain this with some pseudo-python and describe it either with words or mathematical formulas

Reply:
OK. I removed the pseudo-code part. I guess that this one-sentence explanation makes do anyway